charts/sampling/templates/_otelcol_exporter_loadbalancing.river.txt

```diff
@@ -0,0 +1,21 @@
+{{- define "deployment.exporter.loadbalancing" -}}
+otelcol.exporter.loadbalancing "default" {
+  // https://grafana.com/docs/agent/latest/flow/reference/components/otelcol.exporter.loadbalancing/
+  resolver {
+    dns {
+      hostname = "{{ .Release.Name }}-grafana-agent-statefulset.{{ .Release.Namespace }}.svc.cluster.local."
+    }
```

Comment:
Does it make more sense to use the `kubernetes` resolver?
https://grafana.com/docs/agent/latest/flow/reference/components/otelcol.exporter.loadbalancing/#kubernetes-block

Reply:
What are the main benefits it provides over the DNS resolver with a headless service? I think either resolve will work here. Currently this is setup to create a headless service and use the DNS resolver.  

We will likely have to make this resolver configurable however once we add support for multi-cluster setups where users having complex request life-cycles that live across more than a single kubernetes cluster. In this case, they'd need to deploy this sampling chart in each cluster but will have to provide ingress to the second layer, and the exporters will need to be configured to send to this ingress point.

Reply:
Updating to the k8s resolver as the DNS resolver could result in a several second delay after scaling in some cases. I'm also overcomplicating the multi-cluster setup and we can solve this by just adding ingress into the first layer and all clusters would send traces to this ingress.  This should usually not be necessary as requests living across clusters is probably more of an edge case. My assumption is most users will want this helm chart deployed once per cluster.